server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts

```diff
@@ -246,82 +249,191 @@ export default class ProbationPractitionerReferralsController {
+      { userId: res.locals.user.userId }
+    )
+    res.redirect(`/probation-practitioner/referrals/${req.params.id}/cancellation/${draftCancellation.id}/reason`)
+  }
+
+  async backwardsCompatibilityUpdateCancellationReason(req: Request, res: Response): Promise<void> {
     const { user } = res.locals
     const { accessToken } = user.token
+    const referralId = req.params.id
 
-    const sentReferral = await this.interventionsService.getSentReferral(accessToken, req.params.id)
+    const data = await new ReferralCancellationReasonForm(req).data()
+
+    let formError: FormValidationError | null = null
+
+    if (req.method === 'POST') {
+      if (!data.error) {
+        const draftCancellation = await this.draftsService.createDraft<DraftCancellationData>(
+          'cancellation',
+          data.paramsForUpdate,
+          { userId: res.locals.user.userId }
+        )
+
+        return res.redirect(
+          `/probation-practitioner/referrals/${referralId}/cancellation/${draftCancellation.id}/check-your-answers`
+        )
+      }
+
+      res.status(400)
+      formError = data.error
+    }
+
+    const sentReferral = await this.interventionsService.getSentReferral(accessToken, referralId)
     const intervention = await this.interventionsService.getIntervention(
       accessToken,
       sentReferral.referral.interventionId
     )
     const serviceUser = await this.communityApiService.getServiceUserByCRN(sentReferral.referral.serviceUser.crn)
     const cancellationReasons = await this.interventionsService.getReferralCancellationReasons(accessToken)
 
+    // We don’t use this draft, other than to pass it to the presenter for rendering
+    // (remember this controller action is only temporary)
+    const draftCancellation = await this.draftsService.createDraft<DraftCancellationData>(
+      'cancellation',
+      {
+        cancellationReason: null,
+        cancellationComments: null,
+      },
+      { userId: res.locals.user.userId }
+    )
+
     const presenter = new ReferralCancellationReasonPresenter(
+      draftCancellation,
       sentReferral,
       intervention,
       serviceUser,
-      cancellationReasons
+      cancellationReasons,
+      formError
     )
+
+    await this.draftsService.deleteDraft(draftCancellation.id, { userId: res.locals.user.userId })
+
     const view = new ReferralCancellationReasonView(presenter)
 
     return ControllerUtils.renderWithLayout(res, view, serviceUser)
   }
 
-  async submitFormAndShowCancellationCheckAnswersPage(req: Request, res: Response): Promise<void> {
+  private async fetchDraftCancellationOrThrowSpecificError(req: Request, res: Response) {
+    const id = req.params.draftCancellationId
+    const draftCancellation = await this.draftsService.fetchDraft<DraftCancellationData>(id, {
+      userId: res.locals.user.userId,
+    })
+
+    if (draftCancellation === null) {
+      throw createError(500, `Draft cancellation with ID ${id} not found by drafts service`, {
```

Comment:
nit: I think this should throw a 404 (not found) rather than 500?

Reply:
I think semantically that's probably right, but if I remember correctly I wanted to make sure this error shows up in monitoring to see if it's happening in real life, and I think that our alerting doesn't trigger for 404s. Does that sound sensible?

Reply:
yeah that makes sense, perhaps we should do a change to allow for any error with a userMessage to display? 
Although this is another PR...